src/mlpack/core/util/mlpack_main.hpp

```diff
@@ -213,16 +214,25 @@ static const std::string testName = "";
 #include <mlpack/core/util/param.hpp>
 
 #undef BINDING_NAME
-#define BINDING_NAME(NAME) static \
-    mlpack::util::ProgramName \
-    io_programname_dummy_object = mlpack::util::ProgramName(NAME); \
-    namespace mlpack { \
-    namespace bindings { \
-    namespace python { \
-    std::string programName = NAME; \
-    } \
-    } \
-    }
+#ifdef __COUNTER__
+  #define BINDING_NAME(NAME) static \
+      mlpack::util::ProgramName \
+      JOIN(io_programname_dummy_object_, __COUNTER__) = \
+      mlpack::util::ProgramName( \
+      NAME); \
+      mlpack::bindings::python::ChangeProgramName \
+		  JOIN(changename_dummy_, __COUNTER__) = \
+          mlpack::bindings::python::ChangeProgramName(NAME);
```

Comment:
I think as part of #2995, we'll have to use `__COUNTER__` too in all of the binding macros so that more than one `ProgramName` (or `BindingName` as it is called in that branch) can be created in a program.  So, maybe that will mean that we don't need to modify the definition of `BINDING_NAME` here (or `BINDING_USER_NAME` as it will be called).